.github/workflows/publish-ui-components.yml

```diff
@@ -0,0 +1,51 @@
+name: Publish ui-components to NPM Registry
+
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - 'packages/ui-components/**'
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v1
+        with:
+          node-version: '12.x'
+      - name: Install dependencies
+        run: yarn install
+        working-directory: ./packages/ui-components
+      - name: Test
+        run: yarn ci
+        working-directory: ./packages/ui-components
+  publish:
+    runs-on: ubuntu-latest
+    needs: test
+    strategy:
+      matrix:
+        node-version: [12.x]
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v1
+        with:
+          node-version: '12.x'
+          registry-url: 'https://registry.npmjs.org'
+      - name: Install dependencies
+        run: yarn install
+        working-directory: ./packages/ui-components
+      - name: Configure Git User
+        run: |
+          git config user.name "${{ github.actor }}"
+          git config user.email "<>"
+      - name: Bump version (prerelease)
+        run: |
+          yarn version --prerelease
```

Comment:
This was the part I haven't quite figured out. I would like to stick to _something_ like semantic versioning, but I'm not sure how a github action would be able to make that decision. In doing a prerelease version is it your intention that these are developer versions only? So incrementing the version would still happen manually?

Reply:
`yarn version --prerelease` command automatically:
- increments version in `package.json` for example `0.2.6-alpha.0` --> `v0.2.6-alpha.1`. `--prerelease` flag indicates which part to increment. Only major releases have to be done manually.
- Creates tag as the version number

So developers don't need to manually create versions.